### ISSUE TITLE
How to use the model  phoenix-2014t_g2t

### ISSUE THREAD

**Issue by kingbeam** [rzhao-zhsq/CV-SLT#3]
Hello! Thank you for your work. When I was replicating the project, I didn't figure out how to use this file. Could you please give me a hand? Thanks.

**Comment by rzhao-zhsq** [comment]
Plz refer to:

https://github.com/rzhao-zhsq/CV-SLT/blob/1219698b7c4f56fe471f21734811703464cd6b50/modelling/translation.py#L91-L92

and make the necessary modifications or additions to cfg['TranslationNetwork']['load_ckpt'] in the file ```experiments/configs/SingleStream/phoenix-2014t_vs2t.yaml```.


For more detailed instructions, we recommend you to follow the guidelines provided in [MMTLB](https://github.com/FangyunWei/SLRT/blob/main/TwoStreamNetwork/docs/SingleStream-SLT.md). Hope this information is helpful to you.

**Comment by kingbeam** [comment]
Thank you for your reply. I have tried training on phoenix-2014t without modifying any other parameters. I'm using an RTX 4090, but the results I got are quite different from those in the paper. I downloaded two ckpt files and the Mbart_de file from MMTLB and loaded them. Is there anything else I need to do? Thank you. The following are my log contents.
2025-03-13 18:39:53,004 Load Visual Head from pretrained ckpt /home/vs1/CV-SLT-main/pretrained_models/phoenix-2014t_s2g/ckpts/best.ckpt
2025-03-13 18:39:55,139 Initialize translation network from /home/vs1/CV-SLT-main/pretrained_models/Mbart_de
2025-03-13 18:39:55,298 gloss2embed_file /home/vs1/CV-SLT-main/pretrained_models/Mbart_de/gloss_embeddings.bin
2025-03-13 18:39:56,531 Initialize gloss embedding from /home/vs1/CV-SLT-main/pretrained_models/Mbart_de/gloss_embeddings.bin
2025-03-13 18:39:56,547 Loading and Reinitializing Translation network from pretrained ckpt /home/vs1/CV-SLT-main/pretrained_models/phoenix-2014t_g2t/best.ckpt
2025-03-13 18:39:57,499 The content of load_dict:
2025-03-13 18:39:58,438 # Total parameters = 370745670
2025-03-13 18:39:58,438 # Total trainable parameters = 370745670
2025-03-13 18:40:04,042 learning rate recognition_network=0.001
2025-03-13 18:40:04,042 learning rate translation_network=1e-05
2025-03-13 18:40:04,042 learning rate vl_mapper=0.001
2025-03-13 18:40:04,046 Epoch 0, Training examples 7096
2025-03-13 18:42:08,802 {'recognition': {'beam_size': 1}, 'translation': {'length_penalty': 1, 'max_length': 100, 'num_beams': 5}}
2025-03-13 18:42:08,803 Evaluation epoch=0 validation examples #=519
2025-03-13 18:42:59,752 posterior_translation_loss Average:66.7379
2025-03-13 18:42:59,753 prior_translation_loss Average:69.2450
2025-03-13 18:42:59,753 kl_loss Average:0.0794
2025-03-13 18:42:59,754 gkl_loss Average:0.0289
2025-03-13 18:42:59,754 translation_loss Average:136.0912
2025-03-13 18:42:59,754 recognition_loss Average:148.5894

2025-03-13 20:55:12,039 Load model ckpt from experiments/outputs/Variational/phoenix-2014t_vs2t/ckpts/best.ckpt
2025-03-13 20:55:12,039 Evaluate on dev set
2025-03-13 20:55:12,397 {'recognition': {'beam_size': 1}, 'translation': {'length_penalty': 1, 'max_length': 100, 'num_beams': 5}}
2025-03-13 20:55:12,397 Evaluation epoch=26 validation examples #=519
2025-03-13 20:56:41,859 posterior_translation_loss Average:19.2822
2025-03-13 20:56:41,859 prior_translation_loss Average:21.8795
2025-03-13 20:56:41,859 kl_loss Average:1.1225
2025-03-13 20:56:41,859 gkl_loss Average:0.0036
2025-03-13 20:56:41,860 translation_loss Average:42.2877
2025-03-13 20:56:41,860 recognition_loss Average:165.1997
2025-03-13 20:56:41,860 total_loss Average:42.2877
2025-03-13 20:56:42,031 bleu1: 52.28, bleu2: 39.69, bleu3: 31.81, bleu4: 26.48
2025-03-13 20:56:42,184 bleu1: 60.16, bleu2: 47.30, bleu3: 38.85, bleu4: 32.85-Posterior
2025-03-13 20:56:42,184 ROUGE: 51.80
2025-03-13 20:56:42,184 ROUGE: 58.96-Posterior
2025-03-13 20:56:42,186 Evaluate on test set
2025-03-13 20:56:42,611 {'recognition': {'beam_size': 1}, 'translation': {'length_penalty': 1, 'max_length': 100, 'num_beams': 5}}
2025-03-13 20:56:42,612 Evaluation epoch=26 validation examples #=642
2025-03-13 20:58:03,550 posterior_translation_loss Average:17.2331
2025-03-13 20:58:03,551 prior_translation_loss Average:19.7210
2025-03-13 20:58:03,551 kl_loss Average:1.1537
2025-03-13 20:58:03,551 gkl_loss Average:0.0033
2025-03-13 20:58:03,551 translation_loss Average:38.1112
2025-03-13 20:58:03,551 recognition_loss Average:154.8913
2025-03-13 20:58:03,551 total_loss Average:38.1112
2025-03-13 20:58:03,744 bleu1: 52.50, bleu2: 39.95, bleu3: 32.04, bleu4: 26.68
2025-03-13 20:58:03,914 bleu1: 60.86, bleu2: 48.59, bleu3: 40.41, bleu4: 34.55-Posterior
2025-03-13 20:58:03,915 ROUGE: 51.54
2025-03-13 20:58:03,915 ROUGE: 59.73-Posterior

**Comment by kingbeam** [comment]
![Image](https://github.com/user-attachments/assets/5ee55b70-fc54-4d4b-8b81-4d7b0ef8969c)
And the fluctuations of my train/recognition_loss are extremely obvious

**Comment by rzhao-zhsq** [comment]
It seems that the script runs correctly, though the results are not identical as reported in the paper (maybe caused by the hardware/software differences, e,g. 4090 RTX vs TITAN RTX. In my case, the TITAN RTX is a little better than 3090 RTX, and the 3090s are sensitive to some hyper-parameters settings). 

Now, the only one difference is that the recoginition loss in your script makes gradient during training while for vs2t training, the CTC module makes no gradient. Plz make sure the recognition_weight as 0 here if want to purely reproduce the results, and pay attention to the following code blocks:

https://github.com/rzhao-zhsq/CV-SLT/blob/4724269511688075e9b6d5782107fd0a40b2a57e/modelling/model.py#L44

https://github.com/rzhao-zhsq/CV-SLT/blob/4724269511688075e9b6d5782107fd0a40b2a57e/modelling/model.py#L115-L118

https://github.com/rzhao-zhsq/CV-SLT/blob/4724269511688075e9b6d5782107fd0a40b2a57e/training.py#L286-L287

However, i dont think the recoginition_weight makes such difference, maybe you can conduct a few experiments to varify it. So, i recommend you to make some adjustment of hyper-parameters to make full use of 4090s.


![Image](https://github.com/user-attachments/assets/028d3c7e-c8b8-4aa4-a7b0-48385f1a72b4)

![Image](https://github.com/user-attachments/assets/da56ae6e-c8f6-4370-8f8b-5f0d1a7fef24)

**Comment by kingbeam** [comment]
Thank you for your reply. I will try modifying some hyperparameters. Thank you again.
